torch/fx/passes/infra/partitioner.py

```diff
@@ -165,6 +167,18 @@ def merge_single_node(node: Node, id: Optional[int]):
                     # this is a no-op
                     maybe_merge_partition(self_id, other_id)
 
+        # merge the candidates aggressively
+        if self.aggressive_merge:
+            merge_candidates_list = list(set(assignment.values()))
+            while True:
+                size = len(merge_candidates_list)
+                self_id = merge_candidates_list[0]
+                for other_id in merge_candidates_list[1:]:
+                    maybe_merge_partition(self_id, other_id)
+                new_size = len(merge_candidates_list)
```

Comment:
Also, merge_candidates_list is not changed in this loop? 
size would always == new_size.

you can rely on maybe_merge_partition's return to determine if any merge happens.

Reply:
That's a good catch. Thanks.